### DISCUSSION TITLE
Renumbering the cluster/service IPv6 CIDRs

### DISCUSSION THREAD

**Discussion by DerEnderKeks** [k3s-io/k3s#5301] [Q&A]
I'm planning to setup a K3s cluster (with Calico) in my homelab on a residential connection with dual stack. Since my ISP (Deutsche Telekom) hates standards, they hand out a dynamic IPv6 prefix that changes every time my PPPoE connection reconnects. 
As I don't want to use any kind of IPv6 NAT/prefix translation for the cluster if possible, my plan is to renumber the entire cluster ever time my IPv6 prefix changes (only the first 56 bits of the address would need to be changed). This only impacts IPv6, since I have to use NAT for IPv4 anyway.
According to [this issue comment](https://github.com/k3s-io/k3s/issues/2603#issuecomment-735922248) it's not possible to do so. Is this still true? Would it be possible to do it "manually" somehow (without wiping the entire cluster)? I wouldn't mind if some or all of the cluster has to go offline for this to happen, since the uplink was down already anyway when I have to renumber the cluster.  Would it be reasonable to request this to be implemented?

**Comment by brandond** [comment]
There aren't really any documented steps to do this upstream. The apiserver doesn't let you change the CIDR fields on nodes once they're set, so you'd basically have to 
* restart the cluster with the new CIDRs
* delete all the nodes and re-join them

Probably more trouble than it's worth, IMO. I would probably just pick a nonroutable address space for your cluster and enable natv6.
